cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/cube/CubeDeleteQuery.java

```diff
@@ -59,11 +60,12 @@ public CubeDeleteQuery(long startTs, long endTs, int resolution,
    * @param startTs start time of the data selection, in seconds since epoch
    * @param endTs end time of the data selection, in seconds since epoch
    * @param resolution resolution of the aggregations to delete from
-   * @param dimensionValues dimension name, dimension value pairs that define the data selection
+   * @param dimensionValues dimension name, dimension value pairs that define the data selection, note that this should
+   *                        match the prefix of the aggregation group of the metrics.
    * @param measureName name of the measure to delete, {@code null} means delete all
    */
   public CubeDeleteQuery(long startTs, long endTs, int resolution,
-                         Map<String, String> dimensionValues, @Nullable String measureName) {
+                         LinkedHashMap<String, String> dimensionValues, @Nullable String measureName) {
```

Comment:
Why changed to `LinkedHashMap`?

Reply:
Because now the order matters when we delete the metrics, it has to be match the prefix of the aggregation group. Therefore we will need a order preserved map.

Reply:
Can you give an example? And if ordering matters, shouldn't it be a `SortedMap`?

Reply:
Or a list?

Reply:
It will be the prefix of the defined aggregation group. For example, now the application has aggregation group {ns, app, dataset}, the profile has {profilescope, profile, ns, program type, app, program}. Order matters in these aggregation group. When we delete the app, we provide {ns -> nsName, app -> appName} as tags, and both metrics are deleted. We should not delete the profile specific metrics. So we want to change the current behavior to a prefix match. So it is important we provide the tags in correct order. A `SortedMap` does not work since it will automatically sort the map by the key, which is not we want. A list does not work since we want a tag -> tag value key value pair.

Reply:
Changed to use a predicate instead of forcing a LinkedHashMap